spec.md

```diff
@@ -217,26 +217,41 @@ Incompatible changes to the schema SHOULD be reflected by a different URL.
   or application protocol's content-type metadata property. Normative rules
   for the binary mode and the content-type metadata mapping can be found 
   in the respective transport mapping specifications.
-    
+  
 * Constraints:
   * OPTIONAL
   * If present, MUST adhere to the format specified in
     [RFC 2046](https://tools.ietf.org/html/rfc2046)
 * For Media Type examples see [IANA Media Types](http://www.iana.org/assignments/media-types/media-types.xhtml)
 
 ### extensions
+
+### idendityLabels
 * Type: `Map`
-* Description: This is for additional metadata and this does not have a
-  mandated structure. This enables a place for custom fields a producer or
-  middleware might want to include and provides a place to test metadata before
-  adding them to the CloudEvents specification.
-  See the [Extensions](extensions.md) document for a list of possible
-  attributes.
+* Description: This is a place for custom key-value pairs a producer or
+  middleware wants to include to provide additional identity information
+  about the event. The syntax and semantics of each specific label
+  inside this map are open for each event type producer to define,
+  but the keys of this map are of type string and
+  the values are restricted to be of simple types only.
+  For example, a travel request messaging event producer could put
+  "TravelRequestID: 123456" as an identity label for a travel
+  request event, a home IoT device event producer could put
+  "home address: 20 Spruce Street, San Jose, CA" as an identity label
+  for that IoT device event. Identity labels could be used by a
+  serverless platform to correlate this event with other types of
+  events associated with the same serverless application instance.
 * Constraints:
   * OPTIONAL
   * If present, MUST contain at least one entry
 * Examples:
-  * authorization data
+``` JSON
+  properties
```

Comment:
properties being a map is still a bad idea. See my previous comments.

Reply:
OK. Existing spec defines a list of types that can be used by the attributes. Map is the closest one for the need of this attribute, that is why there is further clarification inside the description on the type of the value.

Reply:
if these labels are a map of `LabelKey: LabelValue` then adding metadata on the label is impossible without complex logic to point back to the key.